packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -318,15 +318,11 @@ export type TableProps<
   scrollToRow?: (rows: Row<T>[], data: T[]) => number;
 } & Omit<CommonProps, 'title'>;
 
-const flattenColumns = (columns: Column[]): Column[] => {
-  const flatColumns: Column[] = [];
-  columns.forEach((column) => {
-    flatColumns.push(column);
-    if ('columns' in column) {
-      flatColumns.push(...flattenColumns(column.columns));
-    }
-  });
-  return flatColumns;
+const flattenColumns = <T extends Record<string, unknown>>(
+  columns: Column<T>[],
+): Column<T>[] => {
+  // Return columns itself since we don't support nested columns
+  return columns;
```

Comment:
so do we support the old api or not?

Reply:
The old API is not supported from a types perspective but it still supported from a functionality perspective. So users will see a red line if they pass a top level header with nested columns. But the table will work just as before this PR.

I added a future TODO to stop supporting it from a functionality perspective as well, since I believe in v3, we want to remove the backward compatibility we added from #1072.

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/67e8fcc6bbaa05b18ff87e4a719fdc590b2570ee: Reverted the functionality. But added some stronger types for better type safety.